beetle-core/src/main/java/com/xing/beetle/amqp/MultiPlexingConnection.java

```diff
@@ -92,13 +105,76 @@ public String basicConsume(
       channel.setDefaultConsumer(callback);
       channel.basicQos(this.qosPrefetchSize, this.qosPrefetchCount, this.qosGlobal);
 
-      if (!autoAck) {
-        callback = tagMapping.createConsumerDecorator(callback, channel);
-      }
-      setDefaultConsumer(callback);
+      Consumer callBackForAll =
+          new Consumer() {
+            @Override
+            public void handleConsumeOk(String consumerTag) {
+              callback.handleConsumeOk(consumerTag);
+            }
+
+            @Override
+            public void handleCancelOk(String consumerTag) {
+              callback.handleCancelOk(consumerTag);
+            }
+
+            @Override
+            public void handleCancel(String consumerTag) throws IOException {
+              callback.handleCancel(consumerTag);
+            }
+
+            @Override
+            public void handleShutdownSignal(String consumerTag, ShutdownSignalException sig) {
+              callback.handleShutdownSignal(consumerTag, sig);
+            }
+
+            @Override
+            public void handleRecoverOk(String consumerTag) {
+              callback.handleRecoverOk(consumerTag);
+            }
+
+            @Override
+            public void handleDelivery(
+                String consumerTag, Envelope envelope, AMQP.BasicProperties properties, byte[] body)
+                throws IOException {
+
+              if (callback instanceof MappingConsumer) {
+                MappingConsumer mappingConsumer = (MappingConsumer) callback;
+                Consumer consumer = mappingConsumer.getDelegate();
+                if (consumer.getClass().getName().contains(DefaultChannel.class.getName())) {
+                  // beetle client have the full control
+                  BeetleMessageAdaptor beetleMessageAdaptor =
+                      new BeetleMessageAdaptor(
+                          consumerTags.get(consumerTag), autoAck == true, rejectAndRequeue);
```

Comment:
```suggestion
                          consumerTags.get(consumerTag), autoAck, rejectAndRequeue);
```
I think the comparison to `true` is superfluous.

Reply:
I agree.